src/compiler/binder.ts

```diff
@@ -128,6 +128,8 @@ module ts {
                 case SyntaxKind.FunctionDeclaration:
                 case SyntaxKind.ClassDeclaration:
                     return node.flags & NodeFlags.Default ? "default" : undefined;
+                case SyntaxKind.TypeGuardType:

```

Comment:
Do not like the name of the node/kind; i am not good with naming things though:) may be @JsonFreeman would have ideas.. possibly something with a "Specifier" or "narrowing" in the name would be better.